### DISCUSSION TITLE
Is it possible to also change pitch when using setPlaybackRate()?

### DISCUSSION THREAD

**Discussion by tombburnell** [katspaugh/wavesurfer.js#2673] [Q&A] [answered]
When I use setAudioRate it changes the tempo and maintains the same pitch.
Is it possible to have it change the pitch as well - like a record would. 

I'm currently using backend: MediaElement and for fast loading I think I need to stick with this as it's part of a player on my website for previewing mp3s.

Thanks, Tom

**Comment by aidv** [comment]
No.

But there are ways to make it possible with workarounds.

**Comment by entonbiba** [comment]
@tombburnell you can set your own webaudio filter to achieve this. I'll create a sample soon, but here is something to get started with:

```js
var pitchFilter = wavesurfer.backend.ac.createBiquadFilter();
filter.type = "peaking";
filter.frequency.value = 1000;
filter.Q.value = 1;
filter.gain.value = 10;

wavesurfer.backend.setFilter(pitchFilter);

```

**Comment by katspaugh** [comment, marked as answer]
@tombburnell that's due to how modern browsers deal with the playback speed. From MDN:

> The pitch of the audio is corrected by default. You can disable pitch correction using the [HTMLMediaElement.preservesPitch](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/preservesPitch) property.

See https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/playbackRate

You can see in this example, I've disabled `preservePitch`, and it produces a comical chipmunk voice when you press "octave up": https://codepen.io/katspaugh/pen/xxaYLRa
